job-runner/runner/runner.go

```diff
@@ -106,18 +110,20 @@ TRY_LOOP:
 
 		// Create a JL and send it to the RM.
 		jl := proto.JobLog{
-			RequestId:  r.reqId,
-			JobId:      r.jobId,
-			Name:       r.jobName,
-			Type:       r.jobType,
-			Try:        tryNo,
-			StartedAt:  startedAt,
-			FinishedAt: finishedAt,
-			State:      jobRet.State,
-			Exit:       jobRet.Exit,
-			Error:      errMsg,
-			Stdout:     jobRet.Stdout,
-			Stderr:     jobRet.Stderr,
+			RequestId:   r.reqId,
+			JobId:       r.jobId,
+			Name:        r.jobName,
+			Type:        r.jobType,
+			Try:         tryNo + r.sequenceRetry,
```

Comment:
This strikes me as wrong? Job try should be monotonically increasing. With this, we'll get:
1
2
4
For <job,seq> retry <1,1>, <2,1>, <3,1>

Reply:
Oh actually, it's more like  <1,0>, <2,0>, <3,1> so 1, 2, 4 (mixing tryNo with retry). In any case: seq try/retry counts shouldn't factor into the job try count, which should always be monotonic starting from 1.

Reply:
this is inside a loop of retrying the job. so the try number math was being done outside of the job log. i think the fix is to remove setting try no here and just letting job log handle it

Reply:
fixed this. we chatted offline but for record: job log isn't the source of truth. so to fix, pass in previous job try count and have the traverser be the source of truth.